psi4/src/psi4/libscf_solver/rohf.cc

```diff
@@ -584,15 +584,16 @@ void ROHF::Hx(SharedMatrix x, SharedMatrix ret) {
         // right_ov -= Fb_oi x_iv
         C_DGEMM('N', 'N', occpi[h], virpi[h], doccpi_[h], -0.5, Fbp[0], nmopi_[h], xp[0], virpi[h], 1.0, rightp[0],
                 virpi[h]);
-
-        // Socc terms
-        // left_av += 0.5 * x_oa.T Fb_ov
-        C_DGEMM('T', 'N', soccpi_[h], virpi[h], occpi[h], 0.5, xp[0], virpi[h], (Fbp[0] + doccpi_[h]), nmopi_[h], 1.0,
-                leftp[doccpi_[h]], virpi[h]);
-
-        // right_oa += 0.5 * Fb_op x_ap.T
-        C_DGEMM('N', 'T', occpi[h], soccpi_[h], pvir[h], 0.5, (Fbp[0] + occpi[h]), nmopi_[h],
-                (xp[doccpi_[h]] + soccpi_[h]), virpi[h], 1.0, rightp[0], virpi[h]);
+        if(soccpi_[h]){
+            // Socc terms
+            // left_av += 0.5 * x_oa.T Fb_ov
+            C_DGEMM('T', 'N', soccpi_[h], virpi[h], occpi[h], 0.5, xp[0], virpi[h], (Fbp[0] + doccpi_[h]), nmopi_[h], 1.0,
+                    leftp[doccpi_[h]], virpi[h]);
```

Comment:
I agree with most of the fix, but it still is a bit odd `left[h]` is larger than `docc[h]` as the dimension of left should be `docc + socc x socc + vir`.

Reply:
My understanding is that leftp is allocated to occpi X virpi, which is (docc+socc) X (socc+vir) as you stated.  The result ptr here is offset to `docc`, because this contraction is filling in the `socc` contribution - the stride is correctly stated as `virpi` (`socc`+`docc`).  The memory error resulted from irreps where there are no `socc`, resulting in a null matrix multiply, which BLAS hates.  There are analogous guards above checking for empty `occpi` and `virpi`.  This would've never triggered in C1 symmetry for that reason, which is likely how it slipped through the cracks.